File: Data Structures & Algorithms/duplicate-integer/submission-3.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        if len(nums) == 0:
            return False
        nums_set = set()
        for i in nums:
            check_len = len(nums_set)
            nums_set.add(i)
            if check_len == len(nums_set):
                return True
        return False

File: Data Structures & Algorithms/two-integer-sum/submission-3.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {

        unordered_map<int,int> needed_sums;
        vector<int> res;
        for(int i = 0; i < nums.size(); i++)
        {
            needed_sums[target - nums[i]] = i;
        }

        for(int i = 0; i < nums.size(); i++)
        {
            if (needed_sums.count(nums[i]) && i != needed_sums[nums[i]])
            {
                res = {i, needed_sums[nums[i]]};
                return res;
            }
        }
        return res;
    }
};
